momepy/preprocessing.py

```diff
@@ -740,3 +748,240 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+def _selecting_rabs_from_poly(gdf, circom_threshold = 0.7, perc_area_threshold = 0.85, include_adjacent=True) :
+    """
+    From a GeoDataFrame of polygons, returns a GDF of polygons that are 
+    above the Circular Compaactness threshold.
+    
+    Return
+    ________
+    GeoDataFrames : round abouts and adjacent polygons
+    """
+    # calculate parameters
+    gdf["area"] = gdf.geometry.area
+    gdf["circom"] = CircularCompactness(gdf, "area").series
+    
+    #selecting round about polygons based on compactness
+    rab = gdf[gdf.circom > circom_threshold]
+    #exclude those above the area threshold
+    area_threshold = gdf.area.quantile(perc_area_threshold)
+    rab = rab[rab.area < area_threshold]
+    
+    if include_adjacent == True :
+        #selecting the adjacent areas that are of smaller than itself
+        rab_adj = gpd.sjoin(gdf, rab, predicate = 'intersects')
+        rab_adj = rab_adj[rab_adj.area_right >= rab_adj.area_left]
+        
+        adj_long_polys = []
+        #iterate through original rabs
+        for row in rab.itertuples() :
+            #creating a distance (diameter) for limiting the max dist of adjacent
+            minx, _ , maxx, _ = row.geometry.bounds 
+            rab_diameter = maxx - minx
```

Comment:
I used this as a proxy of the radius of the roundabout. It could also be `maxy - miny`, but assuming most roundabouts are close to a perfect circle those the differences on X and Y axis should be very similar. 
Ideally, one would take the radius from a minimum bounding circle, but I know this is a bit more challenging. 
I will incorporate a small upgrade to this, though.
minx, _ , maxx, _ = row.geometry.bounds 
```suggestion
            minx, maxy , maxx, miny = row.geometry.bounds 
            if (maxx - minx) >= (maxy - miny):
                rab_diameter = maxx - minx
            else:
                rab_diameter = maxy - miny
```